common/src/main/java/com/thekillerbunny/goofyplugin/lua/CollectionAPI.java

```diff
@@ -0,0 +1,300 @@
+            double product = 1.0;
+            for (double v : arr) {
+                product *= v;
+            }
+            return product;
+        } else if (first.isuserdata(FiguraVec2.class)) {
+            FiguraVec2 product = FiguraVec2.of(0, 0);
+            for (int i = 1; i <= tbl.length(); i++) {
+                product.multiply((FiguraVec2) tbl.get(i).checkuserdata(FiguraVec2.class));
+            }
+            return product;
+        } else if (first.isuserdata(FiguraVec3.class)) {
+            FiguraVec3 product = FiguraVec3.of(0, 0, 0);
+            for (int i = 1; i <= tbl.length(); i++) {
+                product.multiply((FiguraVec3) tbl.get(i).checkuserdata(FiguraVec3.class));
+            }
+            return product;
+        } else if (first.isuserdata(FiguraVec4.class)) {
+            FiguraVec4 product = FiguraVec4.of(0, 0, 0, 0);
+            for (int i = 1; i <= tbl.length(); i++) {
+                product.multiply((FiguraVec4) tbl.get(i).checkuserdata(FiguraVec4.class));
+            }
+            return product;
+        } else {
+            throw new LuaError("Unsupported array type");
+        }
+    }
+
+    @LuaWhitelist
+    @LuaMethodDoc(
+            overloads = {
+                    @LuaMethodOverload(
+                            argumentTypes = {LuaTable.class},
+                            argumentNames = {"tbl"}
+                    ),
+
+            },
+            value = "collection.sum"
+    )
+    public Object quotient(@LuaNotNil LuaTable tbl) {
+        if (tbl.length() == 0) {
+            return LuaValue.NIL;
+        }
+
+        LuaValue first = tbl.get(1);
+        if (first.isnumber()) {
+            double[] arr = (double[]) CoerceLuaToJava.coerce(tbl, double[].class);
+            double quotient = 1.0;
+            for (double v : arr) {
+                quotient /= v;
+            }
+            return quotient;
+        } else if (first.isuserdata(FiguraVec2.class)) {
+            FiguraVec2 quotient = FiguraVec2.of(0, 0);
+            for (int i = 1; i <= tbl.length(); i++) {
+                quotient.divide((FiguraVec2) tbl.get(i).checkuserdata(FiguraVec2.class));
+            }
+            return quotient;
+        } else if (first.isuserdata(FiguraVec3.class)) {
+            FiguraVec3 quotient = FiguraVec3.of(0, 0, 0);
+            for (int i = 1; i <= tbl.length(); i++) {
+                quotient.divide((FiguraVec3) tbl.get(i).checkuserdata(FiguraVec3.class));
+            }
+            return quotient;
+        } else if (first.isuserdata(FiguraVec4.class)) {
+            FiguraVec4 quotient = FiguraVec4.of(0, 0, 0, 0);
+            for (int i = 1; i <= tbl.length(); i++) {
+                quotient.divide((FiguraVec4) tbl.get(i).checkuserdata(FiguraVec4.class));
+            }
+            return quotient;
+        } else {
+            throw new LuaError("Unsupported array type");
+        }
+    }
```

Comment:
These always return (the appropriate) `0` for vectors.

Reply:
Here's a patch to default to 1 like doubles do: (idk why it says I replaced the entire code block)
```suggestion
    @LuaWhitelist
    @LuaMethodDoc(
            overloads = {
                    @LuaMethodOverload(
                            argumentTypes = {LuaTable.class},
                            argumentNames = {"tbl"}
                    ),

            },
            value = "collection.sum"
    )
    public Object product(@LuaNotNil LuaTable tbl) {
        if (tbl.length() == 0) {
            return LuaValue.NIL;
        }

        LuaValue first = tbl.get(1);
        if (first.isnumber()) {
            double[] arr = (double[]) CoerceLuaToJava.coerce(tbl, double[].class);
            double product = 1.0;
            for (double v : arr) {
                product *= v;
            }
            return product;
        } else if (first.isuserdata(FiguraVec2.class)) {
            FiguraVec2 product = FiguraVec2.of(1, 1);
            for (int i = 1; i <= tbl.length(); i++) {
                product.multiply((FiguraVec2) tbl.get(i).checkuserdata(FiguraVec2.class));
            }
            return product;
        } else if (first.isuserdata(FiguraVec3.class)) {
            FiguraVec3 product = FiguraVec3.of(1, 1, 1);
            for (int i = 1; i <= tbl.length(); i++) {
                product.multiply((FiguraVec3) tbl.get(i).checkuserdata(FiguraVec3.class));
            }
            return product;
        } else if (first.isuserdata(FiguraVec4.class)) {
            FiguraVec4 product = FiguraVec4.of(1, 1, 1, 1);
            for (int i = 1; i <= tbl.length(); i++) {
                product.multiply((FiguraVec4) tbl.get(i).checkuserdata(FiguraVec4.class));
            }
            return product;
        } else {
            throw new LuaError("Unsupported array type");
        }
    }

    @LuaWhitelist
    @LuaMethodDoc(
            overloads = {
                    @LuaMethodOverload(
                            argumentTypes = {LuaTable.class},
                            argumentNames = {"tbl"}
                    ),

            },
            value = "collection.sum"
    )
    public Object quotient(@LuaNotNil LuaTable tbl) {
        if (tbl.length() == 0) {
            return LuaValue.NIL;
        }

        LuaValue first = tbl.get(1);
        if (first.isnumber()) {
            double[] arr = (double[]) CoerceLuaToJava.coerce(tbl, double[].class);
            double quotient = 1.0;
            for (double v : arr) {
                quotient /= v;
            }
            return quotient;
        } else if (first.isuserdata(FiguraVec2.class)) {
            FiguraVec2 quotient = FiguraVec2.of(1, 1);
            for (int i = 1; i <= tbl.length(); i++) {
                quotient.divide((FiguraVec2) tbl.get(i).checkuserdata(FiguraVec2.class));
            }
            return quotient;
        } else if (first.isuserdata(FiguraVec3.class)) {
            FiguraVec3 quotient = FiguraVec3.of(1, 1, 1);
            for (int i = 1; i <= tbl.length(); i++) {
                quotient.divide((FiguraVec3) tbl.get(i).checkuserdata(FiguraVec3.class));
            }
            return quotient;
        } else if (first.isuserdata(FiguraVec4.class)) {
            FiguraVec4 quotient = FiguraVec4.of(1, 1, 1, 1);
            for (int i = 1; i <= tbl.length(); i++) {
                quotient.divide((FiguraVec4) tbl.get(i).checkuserdata(FiguraVec4.class));
            }
            return quotient;
        } else {
            throw new LuaError("Unsupported array type");
        }
    }
```